### PR TITLE
fix: prevent multiple parent focus events on click

### LINKED ISSUE #4260: [BUG] - Switch inside of CardBody shows blue outline on toggle
### NextUI Version

2.6.3

### Describe the bug

New bug in 2.6.3, if you toggle a switch it will show the blue focus outline but only inside of a CardBody, must be a focus issue. It's a little jarring to have the outline appear when tapping a switch. I'm using global install of @nextui-org/react@2.6.3

Screenshot 2024-12-07 at 2 48 05 AM

### Your Example Website or App

_No response_

### Steps to Reproduce the Bug or Issue

```jsx
import { Card, CardBody, Switch } from "@nextui-org/react"

export default function App() {
    return (
        <Card>
            <CardBody>
                <Switch>
                    Airplane mode
                </Switch>
            </CardBody>
        </Card>
    )
}
```

### Expected behavior

see above

### Screenshots or Videos

_No response_

### Operating System Version

macOS

### Browser

Chrome

### PR DESCRIPTION
**Pull request by Peterl561**
Closes #4260

## 📝 Description

- caused by #4220
- affects `Checkbox`, `Radio`, and `Switch` components
- occurs when parent container is focusable (see [sandbox](https://codesandbox.io/p/github/Peterl561/4260/main?file=%2Fsrc%2FApp.tsx))
- need to call `preventDefault` to prevent parent focus from firing multiple times

## ⛳️ Current behavior (updates)

https://github.com/user-attachments/assets/11655fd6-a81c-4e53-b7ba-dda01fb51d62

## 🚀 New behavior

https://github.com/user-attachments/assets/8ab62565-d06a-4a62-976c-80bd9a6e3f46

## 💣 Is this a breaking change (Yes/No):

No

## Summary by CodeRabbit

- **New Features**
	- Improved focus behavior for checkbox, radio, and switch components to enhance user experience by preventing unnecessary focus events.
- **Bug Fixes**
	- Resolved an issue where focus was incorrectly triggered multiple times on parent elements when interacting with checkbox, radio, and switch components.
- **Tests**
	- Added new test cases to verify correct focus behavior for checkbox, radio, and switch components upon interaction.

### CODE REVIEW

**Comment by macci001** [comment]
Hey @Peterl561, thanks a lot fixing the issue.
The PR looks good to me.

**Review by macci001 — APPROVED** [review]

File: packages/components/checkbox/src/use-checkbox.ts [hunk 1]
```diff
@@ -264,6 +264,16 @@ export function useCheckbox(props: UseCheckboxProps = {}) {
 
   const baseStyles = clsx(classNames?.base, className);
 
+  const mouseProps = useMemo(
+    () => ({
+      onMouseDown: (e: React.MouseEvent<HTMLLabelElement>) => {
```
Comment (macci001) [inline packages/components/checkbox/src/use-checkbox.ts:269]: can you change onMouseDown to onClick?

Comment (macci001) [reply packages/components/checkbox/src/use-checkbox.ts:269]: Same for the other occurrences as well.

File: packages/components/checkbox/src/use-checkbox.ts [hunk 2]
```diff
@@ -264,6 +264,16 @@ export function useCheckbox(props: UseCheckboxProps = {}) {
...
+        // prevent parent from being focused
+        e.preventDefault();
```
[the remark this answers is not in the document]
Comment (macci001) [reply packages/components/checkbox/src/use-checkbox.ts:271]: Same for the other occurrances as well.

**Comment by macci001** [comment]
Handling this issue in #4306

File: packages/components/checkbox/src/use-checkbox.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (macci001) [reply packages/components/checkbox/src/use-checkbox.ts:271]: Please ignore this comment

**Comment by wingkwong** [comment]
closing - fixed in https://github.com/nextui-org/nextui/pull/4311

**Pull request closed without merging**